js/samples/04.ai.b.messageExtensions.AI-ME/src/index.ts

```diff
@@ -106,15 +106,15 @@ const promptManager = new DefaultPromptManager<ApplicationTurnState>(path.join(_
 // Define storage and application
 // - Note that we're not passing a prompt for our AI options as we won't be chatting with the app.
 const storage = new MemoryStorage();
-const app = new Application<ApplicationTurnState>({
-    storage,
-    adapter,
-    botAppId: process.env.MicrosoftAppId,
-    ai: {
+const botAppId = process.env.MicrosoftAppId as string;
+const app = new ApplicationBuilder<ApplicationTurnState>()
```

Comment:
Was there a design decision on why creating a new app is setup by calling multiple functions instead of the following?: 

```typescript
const app = new ApplicationBuilder({
    storage,
    ai: {
        planner,
        promptManager
    },
   // ...
});
```

Reply:
Yup! The main purpose of the `ApplicationBuilder` is to break down the optional config options into separate components- both for readability and to reduce potential bugs. It also matches up with the fluent method chaining.

Quick example of what we drew reference from:
https://www.sohamkamani.com/javascript/builder-pattern/